cel-validation-test/gateway_test.go

```diff
@@ -0,0 +1,470 @@
+/*
```

Comment:
Tiny nit: I'm not really sure where this should live, but I'd rather avoid another top-level directory. I agree that we should probably keep it away from the other validation tests in `/apis` since we probably want to try to make `/apis` as targeted/focused as possible.

Some other options:

1. `/hack/cel-validation`: Most of our presubmits/things resembling tests live in `/hack` right now, which doesn't feel perfect, but it would at least live alongside the rest of the presubmit logic.
2. `/pkg/test/cel`: I think it's likely that we'll have additional go tests over time, this feels like it would be a fairly natural place for them to live.

At this point I think I'm leaning towards 2 here, but open to alternatives.

Reply:
Yeah, this is a great point.

I see that the PR has been updated already with moving everything into `/hack`, but tbh I think that it's better if we put the Go code in `/pkg/test/cel` like @robscott suggested.

Reply:
Oh definitely this should be changed. I was hoping to get feedback regarding this so thanks a lot for the suggestion.

I've choosen option `/hack/cel-validation`. (Because we have unit tests within `pkg` and it wasn't very neat since we run our unit tests using `go test ./pkg/...` which would then also run our e2e tests. This further suggests that perhaps that was not the right choice)

Reply:
Oh, that's also a good point - this is good enough for now, and can be changed later. Thanks.

Reply:
Whoops I noticed our comments collided there :)
I kind totally get why `hack/cel-validation` may not seem ideal, but yeah we could definitely changed it later